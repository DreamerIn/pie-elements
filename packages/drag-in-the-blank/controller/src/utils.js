import reduce from 'lodash/reduce';
import find from 'lodash/find';

export const getAllCorrectResponses = ({ correctResponse, alternateResponses }) => {
  return reduce(correctResponse || {}, (obj, val, key) => {
    obj[key] = [val];

    if (alternateResponses && alternateResponses[key]) {
      obj[key] = [
        ...obj[key],
        ...alternateResponses[key]
      ];
    }

    return obj;
  }, {});
};
