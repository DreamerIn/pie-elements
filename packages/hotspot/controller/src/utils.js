import isEqual from 'lodash/isEqual';
import isEmpty from 'lodash/isEmpty';

export const getCorrectResponse = (choices) => choices
  .filter(c => c.correct)
  .map(c => ({ id: c.id }))
  .sort();

export const isResponseCorrect = (question, session) => {
  const { shapes: { rectangles, polygons } } = question;
  const choices = [...rectangles, ...polygons];
  let correctResponse = getCorrectResponse(choices);

  if (!session || isEmpty(session)) {
    return false;
  }

  if (session.answers && session.answers.length) {
    return isEqual((session.answers || []).sort(), correctResponse);
  } else if (!(correctResponse && correctResponse.length)) {
    return true;
  }
  return false;
};
