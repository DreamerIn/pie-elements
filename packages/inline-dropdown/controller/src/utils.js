import isEqual from 'lodash/isEqual';
import reduce from 'lodash/reduce';
import find from 'lodash/find';

export const isResponseCorrect = (question, session) => {
  if (!question || !session || !session.value) {
    return false;
  }

  let correctResponse = reduce(question.choices, (acc, area, key) => {
    const correctChoice = find(area, c => c.correct);

    acc[key] = correctChoice.value;

    return acc;
  }, {});

  return isEqual((session.value || {}), correctResponse);
};
