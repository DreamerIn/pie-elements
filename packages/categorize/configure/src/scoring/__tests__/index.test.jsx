import { shallow } from 'enzyme';
import React from 'react';

import { Scoring } from '../index';
import { callsOnChange as _callsOnChange } from '../../__tests__/utils';

describe('Scoring', () => {
  let w;
  let onChange = jest.fn();
  const wrapper = extras => {
    const defaults = {
      classes: {},
      className: 'className',
      onChange,
      categories: [],
      correctResponse: [],
      scoring: {
        weighting: {
          enabled: true,
          rules: []
        },
        partial: {
          enabled: true,
          rules: []
        }
      }
    };
    const props = { ...defaults, ...extras };
    return shallow(<Scoring {...props} />);
  };

  describe('snapshot', () => {
    it('renders', () => {
      w = wrapper();
      expect(w).toMatchSnapshot();
    });
  });

  describe('logic', () => {
    const callsOnChange = function() {
      const args = Array.prototype.slice.call(arguments);
      const first = args.shift();
      const startArgs = typeof first === 'string' ? [wrapper(), first] : first;
      _callsOnChange.apply(null, startArgs.concat(args));
    };

    describe('changeWeighting', () => {
      callsOnChange('changeWeighting', { enabled: false, rules: [] }, () => {
        expect(onChange).toBeCalledWith(
          expect.objectContaining({
            weighting: {
              enabled: false,
              rules: []
            }
          })
        );
      });
    });

    describe('changePartial', () => {
      callsOnChange('changePartial', { enabled: false, rules: [] }, () => {
        expect(onChange).toBeCalledWith(
          expect.objectContaining({
            partial: {
              enabled: false,
              rules: []
            }
          })
        );
      });
    });
  });
});
