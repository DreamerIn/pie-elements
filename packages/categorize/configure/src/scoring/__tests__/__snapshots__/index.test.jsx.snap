// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoring snapshot renders 1`] = `
<div
  className="className"
>
  <WithStyles(Weighting)
    categories={Array []}
    onChange={[Function]}
    weighting={
      Object {
        "enabled": true,
        "rules": Array [],
      }
    }
  />
  <WithStyles(Partial)
    categories={Array []}
    correctResponse={Array []}
    onChange={[Function]}
    partial={
      Object {
        "enabled": true,
        "rules": Array [],
      }
    }
  />
</div>
`;
